cloudigrade/analyzer/management/commands/analyze_log.py

```diff
@@ -0,0 +1,158 @@
+"""Management command for analyzing AWS CloutTrail Logs."""
+import json
+
+from django.core.management.base import BaseCommand
+from django.utils.translation import gettext as _
+
+# from account.models import Account, Instance, InstanceEvent
+from util import aws
+
+
+class Command(BaseCommand):
+    """Command to analyze EC2 events from AWS CloudTrail logs.
+
+    Example usage:
+
+        QUEUE_URL="https://sqs.us-east-1.amazonaws.com/372779871274/doppler_customer_1_test_bucket_cloudtrail_events"
+        python manage.py analyze_log $QUEUE_URL
+    """
+
+    help = _('Analyzed AWS CloudTrail logs for EC2 on/off events.')
+
+    def add_arguments(self, parser):
+        """Require an AWS SQS queue URL."""
+        parser.add_argument('queue_url', help=_('AWS SQS Queue URL'))
+
+    def handle(self, *args, **options):
+        """Read SQS Queue for log location, and parse log for events."""
+        queue_url = options['queue_url']
+
+        logs = []
+        parsed_messages = []
+
+        # Get messages off of an SQS queue
+        messages = aws.receive_message_from_queue(queue_url)
+
+        # Parse the SQS messages to get S3 object locations
+        for message in messages:
+            parsed_messages.extend(self._parse_sqs_message(message))
+
+        # Grab the object contents from S3
+        for parsed_message in parsed_messages:
+            bucket = parsed_message[0]
+            key = parsed_message[1]
+            logs.append(aws.get_object_content_from_s3(bucket, key))
+
+        # Parse logs for on/off events
+        for log in logs:
+            if log is None or log == '':
```

Comment:
Maybe simplify to `if not log`?

On second thought, maybe flip the condition around to `if log` and delete the `continue`.

Reply:
I always get myself into weird negative check land when if log is obviously simpler.